src/meshes/triangle_mesh.template.cc

```diff
@@ -36,6 +36,189 @@
 
 namespace oomph
 {
+  //======================================================================
+  /// Build mesh from the structure and geometry of another mesh but
+  /// replace element by a different type (passed in via the template
+  /// parameter as usual).
+  /// A lot of this code is repeated in the construction of the mesh from
+  /// the scaffold mesh.
+  //======================================================================
+  template<class ELEMENT>
+  template<class ORIG_ELEMENT>
```

Comment:
Do we need the second template arg at all? Orig mesh can simply be a Triangle base mesh.

Reply:
Mesh::finite_element_pt(...) does the cast automatically (and dies if it fails).

Reply:
I've removed the second template and added FiniteElement instead however in testing I get a new mesh structure!?
I'm having to do a dynamic cast from my old mesh type to TriangleMesh<FiniteElement>* which might be where the issue originates from. Keeping everything in its original state was why I was double templating.

This is probably more complicated than it needs to be.
I suggest that I close this pull request for now, and we focus on the other one until the purpose of this one is clearer.